`NullnessChecker` false positive with nested conditional operator

Consider the following dummy code:
```sh
$ cat -n Dummy.java 
     1	import java.util.List;
     2	import java.util.Optional;
     3	import org.checkerframework.checker.nullness.qual.NonNull;
     4	
     5	final class Dummy {
     6	  Optional<String> foo(List<String> list) {
     7	    return list.isEmpty() ? Optional.empty() : Optional.of(list.get(0));
     8	  }
     9	
    10	  Optional<Optional<String>> bar(List<String> list) {
    11	    return Optional.of(list.isEmpty() ? Optional.empty() : Optional.of(list.get(0)));
    12	  }
    13	
    14	  Optional<Optional<String>> baz(List<String> list) {
    15	    return Optional.of(
    16	        list.isEmpty() ? Optional.<@NonNull String>empty() : Optional.of(list.get(0)));
    17	  }
    18	
    19	  Optional<Optional<String>> qux(List<String> list) {
    20	    return Optional.of(
    21	        list.isEmpty() ? Optional.empty() : Optional.<@NonNull String>of(list.get(0)));
    22	  }
    23	
    24	  Optional<Optional<String>> quux(List<String> list) {
    25	    return Optional.of(
    26	        list.isEmpty()
    27	            ? Optional.<@NonNull String>empty()
    28	            : Optional.<@NonNull String>of(list.get(0)));
    29	  }
    30	}
```

Compiling this code using the `NullnessChecker` we get the following output:
```
$ javac \
    -J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED \
    -processorpath ~/.m2/repository/org/checkerframework/checker/3.8.0/checker-3.8.0.jar \
    -cp ~/.m2/repository/org/checkerframework/checker-qual
/3.8.0/checker-qual-3.8.0.jar \
    -processor org.checkerframework.checker.nullness.NullnessChecker \
    Dummy.java
Dummy.java:11: error: [return.type.incompatible] incompatible types in return.
    return Optional.of(list.isEmpty() ? Optional.empty() : Optional.of(list.get(0)));
                      ^
  type of expression: @Initialized @NonNull Optional<@Initialized @NonNull Optional<@Initialized @Nullable String>>
  method return type: @Initialized @NonNull Optional<@Initialized @NonNull Optional<@Initialized @NonNull String>>
Dummy.java:15: error: [return.type.incompatible] incompatible types in return.
    return Optional.of(
                      ^
  type of expression: @Initialized @NonNull Optional<@Initialized @NonNull Optional<@Initialized @Nullable String>>
  method return type: @Initialized @NonNull Optional<@Initialized @NonNull Optional<@Initialized @NonNull String>>
Dummy.java:20: error: [return.type.incompatible] incompatible types in return.
    return Optional.of(
                      ^
  type of expression: @Initialized @NonNull Optional<@Initialized @NonNull Optional<@Initialized @Nullable String>>
  method return type: @Initialized @NonNull Optional<@Initialized @NonNull Optional<@Initialized @NonNull String>>
3 errors
```

Note that the checker complains about methods 2, 3 and 4 (i.e. `bar`, `baz` and `qux`) but not about methods 1 and 5 (i.e. `foo` and `quuz`). Somehow wrapping the conditional operator in `Optional.of(...)` requires one to add type hints to both branches.

--- a/checker/tests/nullness/Issue4007.java
+++ b/checker/tests/nullness/Issue4007.java
@@ -1,7 +1,5 @@
 // Test case for issue #4007: https://tinyurl.com/cfissue/4007
 
-// @skip-test until the issue is fixed
-
 import java.util.List;
 import java.util.Optional;
 import org.checkerframework.checker.nullness.qual.NonNull;

--- a/checker/tests/optional/Marks5.java
+++ b/checker/tests/optional/Marks5.java
@@ -14,7 +14,7 @@ public class Marks5 {
   // unless BOTH are empty, in which case return an empty Optional.
 
   Optional<BigDecimal> clever(Optional<BigDecimal> first, Optional<BigDecimal> second) {
-    @SuppressWarnings({"methodref.inference.unimplemented", "methodref.receiver"})
+    @SuppressWarnings("methodref.receiver")
     Optional<BigDecimal> result =
         Stream.of(first, second)
             .filter(Optional::isPresent)

